pkg/handlers/cluster_for_akodeploymentconfig_handler.go

```diff
@@ -32,59 +31,30 @@ func AkoDeploymentConfigForCluster(c client.Client, log logr.Logger) handler.Map
 			return nil
 		}
 		logger := log.WithValues("cluster", cluster.Namespace+"/"+cluster.Name)
-		if SkipCluster(cluster) {
+		if ako_operator.SkipCluster(cluster) {
 			logger.Info("Skipping cluster in handler")
 			return []reconcile.Request{}
 		}
-
-		adcForCluster, err := ListADCsForCluster(ctx, cluster, logger, c)
+		// get akodeploymentconfig object for this cluster
+		adcForCluster, err := ako_operator.GetAKODeploymentConfigForCluster(ctx, c, logger, cluster)
 		if err != nil {
+			logger.Error(err, "failed to get cluster matched akodeploymentconfig object")
 			return []reconcile.Request{}
 		}
-		var requests []ctrl.Request
-		for _, akoDeploymentConfig := range adcForCluster {
-			requests = append(requests, ctrl.Request{
-				NamespacedName: types.NamespacedName{
-					Namespace: akoDeploymentConfig.Namespace,
-					Name:      akoDeploymentConfig.Name,
-				},
-			})
+		requests := []reconcile.Request{}
+		if adcForCluster == nil {
+			logger.V(1).Info("cluster is not selected by any ako deploymentconfig")
+			ako_operator.RemoveClusterLabel(log, cluster)
```

Comment:
Lets do this at the caller. In future someone else may invoke this function without the intention of modifying labels.

Reply:
This is the mapper function to send adc reconcile request when the cluster gets updated, it is just like other controller reconcile functions: 
https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/blob/main/controllers/akodeploymentconfig/akodeploymentconfig_controller.go#L40

Reply:
Got it. But why we have to update cluster-label here? Subsequent adc reconcile requests will update the label on cluster, correct?

Reply:
The problem here is that the mapper function will just call adc controller to reconcile but it didn't tell us which cluster update triggered adc reconcile.
Image this condition:

cluster label updated (should be selected from adc-1 to adc-2, but avi label didn't change) -> mapper function finds the right adc but didn't update the avi label -> adc controller reconcile lists all the clusters this adc matched -> it will filter this specific cluster since it's avi label indicates that it is selected by adc-1.

So we need to update the label in the mapper function because the adc controller is leveraging the label to tell should filter the cluster or not.